Scripts/Tracking/Follow/ObjectFollow.cs

```diff
@@ -0,0 +1,217 @@
+﻿namespace VRTK.Core.Tracking.Follow
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using VRTK.Core.Data.Attribute;
+    using VRTK.Core.Data.Enum;
+    using VRTK.Core.Process;
+    using VRTK.Core.Tracking.Follow.Modifier;
+
+    /// <summary>
+    /// The ObjectFollow mirrors the transform properties of another object based on the given FollowModifier.
+    /// </summary>
+    public class ObjectFollow : SourceTargetProcessor
+    {
+        [Header("Object Follow Settings")]
+
+        [UnityFlag]
+        [Tooltip("The transform properties to follow.")]
+        public TransformProperties follow;
+        [Tooltip("The follow modifier mechanic to apply.")]
+        public FollowModifier followModifier = null;
+
+        /// <summary>
+        /// The ObjectFollowUnityEvent emits an event with a source Transform and a target Transform.
+        /// </summary>
+        [Serializable]
+        public class ObjectFollowUnityEvent : UnityEvent<Transform, Transform> { };

```

Comment:
Code convention: Empty curly braces on same line or on multiple ones? I'd like to do the same for methods, too.

Reply:
Hmm tough one, the default formatting is new lines for curly braces, but as this is kinda more like an event declaration it reads nicer on one line.

Reply:
ok lets go with multi line everywhere then, agreed?